app/src/main/java/org/oppia/app/player/state/hintsandsolution/HintsAndSolutionFragmentPresenter.kt

```diff
@@ -69,10 +77,16 @@ class HintsAndSolutionFragmentPresenter @Inject constructor(
       it.viewModel = this.viewModel
       it.lifecycleOwner = fragment
     }
-    subscribeToCurrentState()
+
+    if (isInTrainMode) {
+      subscribeToCurrentQuestion()
+    } else {
+      subscribeToCurrentState()
+    }
```

Comment:
Not sure, if this is correct or not, meaning if we should depend on its parent or not. While having conversation with @BenHenning he mentioned that the `StatePlayerRecyclerViewAssembler` should be improvised in such a way that it does not depend on question player or exploration player and I was thinking if that applies here also. 
Will assign this to @BenHenning for better review on approach.

Reply:
Thanks Rajat. I think we want to avoid specializing this fragment to each place it can be located, but I can see why it's challenging in this case.

One alternative might be to move the common hints & solutions logic to a common fragment, and have wrapper fragments for explorations & questions that keep things specialized. Consider this fragment hierarchy:

- ExplorationFragment
  - HintsAndSolutionExplorationManagerFragment
    - HintsAndSolutionDialogFragment

and


- QuestionFragment
  - HintsAndSolutionQuestionManagerFragment
    - HintsAndSolutionDialogFragment

Thoughts?

Reply:
Yeah this can be done, if we want to have specific files for each feature.

Reply:
> Thanks Rajat. I think we want to avoid specializing this fragment to each place it can be located, but I can see why it's challenging in this case.
> 
> One alternative might be to move the common hints & solutions logic to a common fragment, and have wrapper fragments for explorations & questions that keep things specialized. Consider this fragment hierarchy:
> 
> * ExplorationFragment
>   
>   * HintsAndSolutionExplorationManagerFragment
>     
>     * HintsAndSolutionDialogFragment
> 
> and
> 
> * QuestionFragment
>   
>   * HintsAndSolutionQuestionManagerFragment
>     
>     * HintsAndSolutionDialogFragment
> 
> Thoughts?

@BenHenning @veena14cs This approach sounds correct to me, as it will still keep the major logic portion independent of its parent. Thanks.